clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -324,32 +329,54 @@ static bool selectBfloatLibs(const llvm::Triple &Triple, const Compilation &C,
       }
     }
 
-    UseNative = false;
-
-    // Check for intel_gpu_pvc as the target
-    if (Arg *SYCLTarget = Args.getLastArg(options::OPT_fsycl_targets_EQ)) {
-      if (SYCLTarget->getValues().size() == 1) {
-        StringRef SYCLTargetStr = SYCLTarget->getValue();
-        if (SYCLTargetStr == "intel_gpu_pvc")
-          UseNative = true;
-      }
-    }
-
-    auto checkBF = [](StringRef Device) {
-      return Device.starts_with("pvc") || Device.starts_with("ats");
-    };
+    // We need to select fallback/native bfloat16 devicelib in AOT compilation
+    // targetting for Intel GPU devices. Users have 2 ways to apply AOT,
+    // 1). clang++ -fsycl -fsycl-targets=spir64_gen -Xs "-device pvc,...,"
+    // 2). clang++ -fsycl -fsycl-targets=intel_gpu_pvc,...
+    // We assume that users will only apply either 1) or 2) and won't mix the
+    // 2 ways in their compiling command.
```

Comment:
What about this case?

`clang++   --offload-new-driver  -fsycl  --offload-arch=bdw`

Reply:
Hi, @srividya-sundaram 
Thanks for pointing out this, this PR didn't consider new offload driver, I will update it.
Thanks very much.